src/WebJobs.Script.WebHost/Management/InstanceManager.cs

```diff
@@ -80,21 +84,47 @@ public async Task<string> ValidateContext(HostAssignmentContext assignmentContex
         {
             _logger.LogInformation($"Validating host assignment context (SiteId: {assignmentContext.SiteId}, SiteName: '{assignmentContext.SiteName}')");
 
+            string error = null;
+
+            try
+            {
+                await ValidateContextInternal(assignmentContext);
+            }
+            catch (Exception e)
+            {
+                _logger.LogError(e.ToString());
+                error = "Invalid zip url specified";
```

Comment:
I don't understand why you pushed all the logic out to a ValidateContextInternal method. What's the benefit? It also means that your error message here isn't really correct - ValidateContext validates the entire context and may fail for reasons other than an invalid zip in the future.

So I'd say just keep the logic inlined as it was before which is the simplest way to deal with the issue.